### PR TITLE
Add squashfs

### LINKED ISSUE #110: SquashFS support
Feature request: add output support for using squashfs as the output archive format.

I would like to use some environments without unpacking.  So far it seems to work fine as long as I either squash the environment directly (but it has to be created in the same path that will be used for mounting), or if I pack with a specified destination, but that requires the extra step of untarring just to squash it.

### PR DESCRIPTION
**Pull request by siboehm**
closes #110

### CODE REVIEW

**Comment by xhochy** [comment]
@siboehm Thanks for doing this 👍 Can you add some documentation on how to use these environments?

**Comment by siboehm** [comment]
py2.7 is failing due to the code using some additions to the `os` interface that were added in py3.3. #173

File: docs/source/squashfs.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+These filesystems can then be mounted directly, without decompressing them first.
+This allows benefiting from compressed storage, without the decompression step necessary for the
+``zip`` and ``tar.gz`` formats.
+
+Packing
+-------
+Packing environments into SquashFS works on MacOS and Linux.
+You will need `squashfs-tools <https://github.com/plougher/squashfs-tools>`_, more specifically
+the ``mksquashfs`` command.
+``squashfs-tools`` can be installed via ``conda install -c conda-forge squashfs-tools``.
```
Comment (xhochy) [inline docs/source/squashfs.rst:15]: This is also often already available by the Linux distribution of the user (and probably the required way to install that if you use `conda-pack` from `defaults`). Thus: Can you add which Ubuntu packages are needed? This is probably the most common platform.

File: docs/source/squashfs.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+- >7: ``xz``
+
+Making the unpacked environment writeable
+-----------------------------------------
+
+SquashFS is a read-only filesystem.
+Sometimes the unpacked environment needs to be writeable on the target machine, for example to install
+more packages.
+A good way to do this is to use `Union mounting <https://en.wikipedia.org/wiki/Union_mount>`_ to
+add a writeable layer on top of the Read-only SquashFS.
```
Comment (xhochy) [inline docs/source/squashfs.rst:87]: Could you add an example here? I guess users will run into this quite fast (sadly).

**Comment by xhochy** [comment]
@siboehm Can you take care of signing the CLA, please?

**Comment by siboehm** [comment]
I signed the CLA on Friday and got a confirmation from echosign.

**Comment by dbast** [comment]
@anaconda-issue-bot check

**Comment by xhochy** [comment]
@mcg1969 @chenghlee Can you add @siboehm to https://github.com/conda/clabot-config/blob/master/.clabot after you have verified the CLA?

**Comment by dbast** [comment]
@anaconda-issue-bot check

**Comment by dbast** [comment]
This looks very good, CLA bot is also happy.

Would prefer merging this first to not have failing CI on master https://github.com/conda/conda-pack/pull/174

**Comment by xhochy** [comment]
@dbast Ah, cool! Didn't know that you also know have these superpowers :)

**Review by xhochy — APPROVED** [review]

**Comment by dbast** [comment]
This now requires a rebase / fixing the conflicts in .github/workflows/main.yml

**Comment by siboehm** [comment]
Rebased, should by good to go.

**Pull request merged**